src/SupraLib/CMakeLists.txt

```diff
@@ -64,7 +64,7 @@ IF(SUPRA_CUDA)
 	MESSAGE(STATUS "Found Cuda Version " ${CUDA_VERSION_STRING})
 	set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-Wno-deprecated-gpu-targets;--use_fast_math;--default-stream=per-thread;-lineinfo")
 	IF(CUDA_VERSION_STRING VERSION_GREATER "8.0")
-		set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};--Wno-deprecated-declarations")
+		set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};--Wno-deprecated-declarations;-Xcompiler=-fPIC")
```

Comment:
This might be necessary, but we should only enable it when "CMAKE_POSITION_INDEPENDENT_CODE" is set

Reply:
Yes, that would be better.. but it looks like CMAKE_POSITION_INDEPENDENT_CODE is by default false, and POSITION_INDEPENDENT_CODE is the per-target property that becomes true by default for shared objects.

Should we add this if BUILD_SHARED is true?